src/lune/builtins/datetime/mod.rs

```diff
@@ -0,0 +1,222 @@
+use chrono::Month;
+use mlua::prelude::*;
+
+pub(crate) mod builder;
+pub(crate) mod date_time;
+
+use self::{
+    builder::DateTimeBuilder,
+    date_time::{DateTime, TimestampType, Timezone},
+};
+use crate::lune::util::TableBuilder;
+
+// TODO: Proper error handling and stuff
+
+pub fn create(lua: &'static Lua) -> LuaResult<LuaTable> {
+    TableBuilder::new(lua)?
+        .with_function("now", |_, ()| Ok(DateTime::now()))?
+        .with_function("fromUnixTimestamp", |lua, timestamp: LuaValue| {
+            let timestamp_cloned = timestamp.clone();
+            let timestamp_kind = TimestampType::from_lua(timestamp, lua)?;
+            let timestamp = match timestamp_kind {
+                TimestampType::Seconds => timestamp_cloned.as_i64().ok_or(LuaError::external("invalid float integer timestamp supplied"))?,
+                TimestampType::Millis => {
+                    let timestamp = timestamp_cloned.as_f64().ok_or(LuaError::external("invalid float timestamp with millis component supplied"))?;
+
+                    ((((timestamp - timestamp.fract()) as u64) * 1000_u64) // converting the whole seconds part to millis
+                    // the ..3 gets a &str of the first 3 chars of the digits after the decimals, ignoring
+                    // additional floating point accuracy digits
+                        + (timestamp.fract() * (10_u64.pow(timestamp.fract().to_string().split('.').collect::<Vec<&str>>()[1][..3].len() as u32)) as f64) as u64) as i64
```

Comment:
This is very confusing to read and has many type casts in it, which could panic at runtime. Can you refactor this to be more readable and try to remove the string conversion + casts?

Reply:
Maybe I could use `try_into` instead? I'll look at what I can do.

Reply:
Marking this as resolved since there's not much that can be done.